test/src/main/java/org/axonframework/test/aggregate/AggregateTestFixture.java

```diff
@@ -715,6 +715,13 @@ public IdentifierValidatingRepository(Repository<T> delegate) {
             return aggregate;
         }
 
+        @Override
+        public Aggregate<T> loadOrCreate(String aggregateIdentifier,Callable<T> factoryMethod) throws Exception{
+            CurrentUnitOfWork.get().onRollback(u -> this.rolledBack = true);
+            aggregate = delegate.loadOrCreate(aggregateIdentifier,factoryMethod);
```

Comment:
Darn it, shame we missed this! However, would you mind adding the following line here too:
```
validateIdentifier(aggregateIdentifier, aggregate);
```